app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt

```diff
@@ -23,8 +29,30 @@ class RocketListFragment : Fragment() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        viewModel.rocketLiveData.observe(this) { rockets ->
-            binding.rocketList.adapter = RocketListAdapter(rockets)
+        lifecycleScope.launch {
```

Comment:
podle me tohle muze zpusobovat problemy a bylo by lepsi pouzit
viewLifecycleOwner.lifecycleScope.launch {  } v onViewCreated. Konkretne se muze stat ze binding bude null a appka crashne.

Reply:
Zkontroloval jsem dokumentaci, bude to v poradku. `repeatOnLifecycle` se zrusi pri prechodu do STOP stavu
> repeatOnLifecycle launches the block in a new coroutine every time the
lifecycle is in the STARTED state (or above) and cancels it when it's STOPPED.

https://developer.android.com/topic/libraries/architecture/coroutines#restart

Jeste je mozne vyuzit o jednu uroven zanoreni mene s Flow.flowWithLifecycle() https://developer.android.com/reference/kotlin/androidx/lifecycle/package-summary#flowwithlifecycle, tak to vyuziju a trochu rfc.